### ISSUE TITLE
Suitesparse library not copied over to build directory

### ISSUE THREAD

**Issue by vtraag** [microsoft/vcpkg#14203]
I am trying to build a library (`igraph`) that depends on `suitesparse` (`cxsparse`), for which I am trying to use `vcpkg`. Installing `suitesparse` using `vcpkg` works perfectly fine, and the package is also correctly located when building the other library. However, for some reason the appropriate DLL is not copied over to the build directory (it is my understanding that this is normally done by `applocal.ps1`?). In this case, this concerns `libcxsparsed.dll`. Copying over `libcxsparsed.dll` manually works well, but preferably this would not need to be done. Other dependencies from `vcpkg` are correctly copied over.

- OS: Windows
- Compiler: MSVC14.27.29110
- Build type: shared debug library.

For some more background information see https://github.com/igraph/igraph/issues/1492.

**Comment by JackBoosY** [comment]
Related PR https://github.com/microsoft/vcpkg/issues/13797

**Comment by vtraag** [comment]
To be clear: this question does not concern the installation phase, but merely the build phase. For other packages (e.g. `libxml2`) the relevant `dll` is copied to the *build* directory of `igraph`. For `suitesparse` this is not the case for some reason.

**Comment by ras0219** [comment]
Sorry for the long wait for a response!

vcpkg uses `dumpbin.exe /dependents` to analyze the dependency chains; you can use this on your produced executables and DLLs to see where the copying may be going wrong (it's part of the MSVC toolset and available in a developer prompt). Usually, an issue like this is caused by either:
1. Improper library linkage (linking to the debug library instead of the release library or vice versa)
2. `LoadLibrary()` calls in the binary. We can only see the static loader table, so we can't automatically copy dynamic "plugin" dependencies.

**Comment by JackBoosY** [comment]
I think this issue can be resolved by PR https://github.com/microsoft/vcpkg/pull/13011 (since you use the cmake project).
Can you confirm that?

**Comment by vtraag** [comment]
Note that the issue did not concern the installation phase, just the build phase. We test the package after building, but it then misses the suitesparse dependency. PR #16210 that I just pushed should fix this problem.

**Comment by PhoebeHui** [comment]
@vtraag, thanks for the PR! The PR looks good to me.

I'm concerned if the PR fix this issue, I installed suitesparse locally, and double confirmed libcxsparsed.dll was copied to installed folder correctly, moreover vcpkg_copy_pdbs() only copy the pdb files. Could you double confirm if the PR fix the issue?

**Comment by vtraag** [comment]
Sorry for the long delay, I haven't had time yet to double check. Hopefully I will get around to that shortly.

**Comment by vtraag** [comment]
It seem that #16210 does not solve the issue unfortunately, presumably the correct library remained in the build directory when I tested this possibility.

I have now traced the problem in more detail. In particular, in `applocal.ps1`, the dependents are traced and copied over. In particular, when compiling in debug mode, it cannot find the dependent:
```
VERBOSE: Resolving C:/Users/traagva1/build/igraph-shared/src/igraph.dll...
VERBOSE:   libcxsparse.dll: C:/Users/traagva1/src/vcpkg/installed/x64-windows/debug/bin\libcxsparse.dll not found
```
That is because CMake finds the library `C:/Users/traagva1/src/vcpkg/installed/x64-windows/lib/libcxsparse.lib`. The library contained in the debug directory is called `libcxsparsed.dll` (note the `d` suffix). When compiling in Release mode, this is not a problem, since the correct `libcxsparse.dll` is then picked up.

The proper solution would be to ensure that in Debug configuration the debug libraries are being picked up. It turns out that the `FindCXSparse.cmake` included in `igraph` does not pick this up. I will try to amend that `cmake` file to accomodate the current setup from `vcpkg` also.
